sdjournal/journal.go

```diff
@@ -100,7 +302,19 @@ func NewJournal() (*Journal, error) {
 // in a given directory. The supplied path may be relative or absolute; if
 // relative, it will be converted to an absolute path before being opened.
 func NewJournalFromDir(path string) (*Journal, error) {
-	path, err := filepath.Abs(path)
+	h, err := dlopen.GetHandle(libsystemdNames)
+	if err != nil {
+		return nil, err
+	}
+
+	path, err = filepath.Abs(path)
+	if err != nil {
+		return nil, err
+	}
+
+	j := &Journal{lib: h}
+
+	sd_journal_open_directory, err := j.getFunction("sd_journal_open_directory")
 	if err != nil {
 		return nil, err
```

Comment:
You're leaking the handle here right? Need to close h if returned err is non-nil?